images/ipfailover/keepalived/lib/config-generators.sh

```diff
@@ -116,59 +114,31 @@ function generate_track_script() {
 function generate_mucast_options() {
   echo ""
 
-  if [ -n "$MULTICAST_SOURCE_IPADDRESS" ]; then
-    echo "    mcast_src_ip $MULTICAST_SOURCE_IPADDRESS"
+  if [[ -n "${MULTICAST_SOURCE_IPADDRESS}" ]]; then
+    echo "    mcast_src_ip ${MULTICAST_SOURCE_IPADDRESS}"
   fi
 
-  if [ -n "$UNICAST_SOURCE_IPADDRESS" ]; then
-    echo "    unicast_src_ip $UNICAST_SOURCE_IPADDRESS"
+  if [[ -n "${UNICAST_SOURCE_IPADDRESS}" ]]; then
+    echo "    unicast_src_ip ${UNICAST_SOURCE_IPADDRESS}"
   fi
 
-  if [ -n "$UNICAST_PEERS" ]; then
+  if [[ -n "${UNICAST_PEERS}" ]]; then
     echo ""
     echo "    unicast_peer {"
 
-    for ip in $(echo "$UNICAST_PEERS" | tr "," " "); do
-      echo "        $ip"
+    OLD_IFS=$IFS
+    IFS=","
+    upeers=(${UNICAST_PEERS})
+    IFS=$OLD_IFS
+    for ip in ${upeers[@]} ; do
```

Comment:
You don't have to make an array, the following will work:

```sh
OLD_IFS="${IFS}"
for ip in ${UNICAST_PEERS}; do
    # ...
done
IFS="${OLD_IFS}"
```

But if you do want to use the array, you must quote the array expansion:
```sh
for ip in "${upeers[@]}"; do
```